core/cluster.cxx

```diff
@@ -276,6 +289,64 @@ class cluster_impl : public std::enable_shared_from_this<cluster_impl>
     do_open(std::move(handler));
   }
 
+#ifdef COUCHBASE_CXX_CLIENT_COLUMNAR
+  void open_in_background(couchbase::core::origin origin,
+                          utils::movable_function<void(std::error_code)>&& handler)
+  {
+    if (stopped_) {
+      return handler(errc::network::cluster_closed);
+    }
+    if (background_open_started_) {
+      CB_LOG_DEBUG("Background open already started for cluster, id: \"{}\"", id_);
+      return handler({});
+    }
+    if (origin.get_nodes().empty()) {
+      stopped_ = true;
+      work_.reset();
+      return handler(errc::common::invalid_argument);
+    }
+
+    origin_ = std::move(origin);
+    CB_LOG_DEBUG(R"(open cluster, id: "{}", core version: "{}", {})",
+                 id_,
+                 couchbase::core::meta::sdk_semver(),
+                 origin_.to_json());
+    // ignore the enable_tracing flag if a tracer was passed in
+    if (nullptr != origin_.options().tracer) {
+      tracer_ = origin_.options().tracer;
+    } else {
+      if (origin_.options().enable_tracing) {
+        tracer_ = std::make_shared<tracing::threshold_logging_tracer>(
+          ctx_, origin_.options().tracing_options);
+      } else {
+        tracer_ = std::make_shared<tracing::noop_tracer>();
+      }
+    }
+    tracer_->start();
+    // ignore the metrics options if a meter was passed in.
+    if (nullptr != origin_.options().meter) {
+      meter_ = origin_.options().meter;
+    } else {
+      if (origin_.options().enable_metrics) {
+        meter_ = std::make_shared<metrics::logging_meter>(ctx_, origin_.options().metrics_options);
+      } else {
+        meter_ = std::make_shared<metrics::noop_meter>();
+      }
+    }
+    meter_->start();
+    session_manager_->set_tracer(tracer_);
+    session_manager_->set_dispatch_timeout(origin_.options().dispatch_timeout);
```

Comment:
We might need to also set the dispatch timeout in the standard `open()` as well. Right now, if you build the SDK with `COUCHBASE_CXX_CLIENT_COLUMNAR` but use `open` instead of `open_in_background` all operations time out because the dispatch timeout is 0.

Reply:
Hmmm, interesting.  Why would someone build w/ `COUCHBASE_CXX_CLIENT_COLUMNAR` and _not_ use `open_in_background()`?  That would violate (seems like maybe a harsh word 😆) the RFC right?

Reply:
In my opinion, the code hidden behind `COUCHBASE_CXX_CLIENT_COLUMNAR` should be an extension to the existing core, so building with it ideally should not break existing non-columnar behaviour. I think if we don't want `open` to be used with `COUCHBASE_CXX_CLIENT_COLUMNAR` maybe instead of having a separate `open_in_background` we should just have two versions of `open` with `#ifdef #else #endif`?

The reason I raised this is because our testing setup tries to do `cluster_describe` request in order to get the server version etc., and that fails with a timeout because of this at the moment.

Reply:
Lets think about it as "experimental" code for early preview. Now we have CI checking that both versions are buildable, and after merge we can iterate on the test coverage with the final goal of fusing all code paths into single version.